lib/src/server.dart

```diff
@@ -170,22 +173,21 @@ class Server {
         'result': result,
         'id': request['id']
       };
-    }).catchError((error, stackTrace) {
-      if (error is! RpcException) {
-        error = new RpcException(
-            error_code.SERVER_ERROR, getErrorMessage(error), data: {
-          'full': error.toString(),
-          'stack': new Chain.forTrace(stackTrace).toString()
-        });
-      }
-
+    } catch (error, stackTrace) {
       if (error.code != error_code.INVALID_REQUEST &&
```

Comment:
However that drops the `error.code != error_code.INVALID_REQUEST`. Before, if `error` were an `RpcException` with code `INVALID_REQUEST`, it would be serialised and returned; now null may be returned instead. To preserve that logic, you would have to change the first check to:

```
if ((error is RpcException && error.code != error_code.INVALID_REQUEST || error is! RpcException) &&
      !request.containsKey('id')) {
    return null;
}
```